src/compiler/crystal/semantic/method_lookup.cr

```diff
@@ -121,6 +121,7 @@ module Crystal
           macro_owner = item.def.macro_owner?
           context.defining_type = macro_owner if macro_owner
           context.def_free_vars = item.def.free_vars
+          context.free_vars = nil
```

Comment:
Maybe try clear? That way we don't allocated new hashes all the time. Plus it can remain as a getter